### PR TITLE
feat(Search input): Added support for submit button

### LINKED ISSUE #5963: Search input with submit button
https://github.com/patternfly/patternfly/pull/4180

A new variation of the search input that includes a submit button

cc @mcarrano

### PR DESCRIPTION
**Pull request by tlabaj**
**What**: Closes #5963

### CODE REVIEW

**Review by mcarrano — COMMENTED** [review]
Looks good @tlabaj . Is this coded so that either pressing enter or clicking the new submit button will trigger the search?

**Comment by nicolethoen** [comment]
the a11y tests are failing because `aria-expanded="true"` is being applied to the nodes that don't have children and are not actually expandable. That may be a bug related to the `defaultAllExpanded` prop. It should only use `aria-expanded` if it has children to expand/collapse.

File: packages/react-core/src/components/SearchInput/SearchInput.tsx [hunk 1]
```diff
@@ -322,6 +322,17 @@ const SearchInputBase: React.FunctionComponent<SearchInputProps> = ({
               )}
             </span>
           )}
+          {!!onSearch && (
+            <Button
```
Comment (mcoker) [inline packages/react-core/src/components/SearchInput/SearchInput.tsx:326]: This should be a child of the `<InputGroup>`, not `<div className={css(styles.searchInputBar)}>`

**Review by mcoker — CHANGES_REQUESTED** [review]
Looks great! Just one change to match the core example - https://pf4.patternfly.org/components/search-input/#with-submit-button

**Review by nicolethoen — DISMISSED** [review]
Once the tests pass and the structure is updated, looks good to me!

**Comment by tlabaj** [comment]
> Looks good @tlabaj . Is this coded so that either pressing enter or clicking the new submit button will trigger the search?
@mcarrano Katerina added that in PR #6145

**Review by kmcfaul — APPROVED** [review]

**Review by mcarrano — APPROVED** [review]
Looks great. Thanks @tlabaj !

**Review by mcoker — APPROVED** [review]
LGTM!!

**Review by evwilkin — APPROVED** [review]
LGTM

**Review by dlabrecq — COMMENTED** [review]
LGTM

**Pull request merged**